accord-core/src/main/java/accord/primitives/Txn.java

```diff
@@ -96,34 +108,64 @@ enum Kind
          *
          * Invisible to other transactions.
          */
-        ExclusiveSyncPoint,
+        ExclusiveSyncPoint('X'),
+
+        /**
+         * Used for local book-keeping only, not visible to any other replica or directly to other transactions.
+         * This is used to create pseudo transactions that take the place of dependencies that will be fulfilled by a bootstrap.
+         */
+        LocalOnly;
 
-        LocalOnly
-        ;
+        Kind()
+        {
+            this.shortName = name().charAt(0);
+        }
+
+        Kind(char shortName)
+        {
+            this.shortName = shortName;
+        }
 
         public enum Kinds implements Predicate<Kind>
         {
-            Ws, RorWs, WsOrSyncPoint, SyncPoints, AnyGlobal, Any;
+            Nothing,
+            Ws,
```

Comment:
Ws is not a durable write? Do they witness ephemeral reads?

Reply:
I don't follow. Ws just means writes.

Reply:
This just seems like it should be included in the comment below (and be below the comment).

Reply:
I’m confused. Ws is the set of write operations - that doesn’t include any reads, only writes. RorWs needs to specify it doesn’t see one type of read.